portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,21 +38,31 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final UserService userService = UserServiceFactory.getUserService();
   private static Gson gson = new Gson();
 
   /**
    * Gives response containing a single Json List with all stored comments from datastore
    * List sorted from oldest to newest comment
    */
-   //TODO return timestamp and author alongside comment text
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+    Query query = new Query("Comment").addSort("timestamp-ms", SortDirection.ASCENDING);
     PreparedQuery results = datastore.prepare(query);
 
-    List<String> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      comments.add((String) entity.getProperty("text"));
+    List<Comment> comments = new ArrayList<>();
+    for (Entity commentEn : results.asIterable()) {
+      String authorId = (String) commentEn.getProperty("author-id");
+      // TODO query all user entities at once
+      Entity authorEn = UserInformationServlet.getUserEntity(authorId);
+      if(authorEn != null) {
+        comments.add(Comment.builder()
+            .commentText((String) commentEn.getProperty("comment-text"))
+            .authorName((String) authorEn.getProperty("display-name"))
+            .timestampMs((long) commentEn.getProperty("timestamp-ms"))
+            .build()
+        );
+      }
```

Comment:
Do we want to log some type of error in the case where authorEn == null?  As some type of flag that there is some issue with the data?

Reply:
I'm not sure of a good way to handle this to be honest. If someone deletes their account perhaps their comments should go along with them? And then if we run into comments without a valid account-id then we know to discard this comment. I think that's what I'll change it to, but I'm open to other solutions.